# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../mol/dhcp-fixes.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---
diff -ur mol-0.9.70/Doc/config/dhcpd-mol.conf mol-0.9.70-fixed/Doc/config/dhcpd-mol.conf
--- mol-0.9.70/Doc/config/dhcpd-mol.conf	2004-03-27 12:04:27.000000000 +0100
+++ mol-0.9.70-fixed/Doc/config/dhcpd-mol.conf	2005-02-20 19:05:39.924503728 +0100
@@ -12,7 +12,7 @@
 #option domain-name-servers		10.0.0.1, 10.0.0.2;
 
 # Newer dhcpd versions require the following line
-#ddns-update-style	ad-hoc;
+ddns-update-style	ad-hoc;
 
 # Support up to 10 sessions (each with its own tun interface)
 
diff -ur mol-0.9.70/Doc/config/tunconfig mol-0.9.70-fixed/Doc/config/tunconfig
--- mol-0.9.70/Doc/config/tunconfig	2004-03-27 12:04:27.000000000 +0100
+++ mol-0.9.70-fixed/Doc/config/tunconfig	2005-02-20 19:15:13.503306496 +0100
@@ -137,7 +137,10 @@
 IFACES=`netstat -i | sed -n -e 's/^\(tun[0-9]\).*/\1/gp'`
 
 if [ "$IFACES" ] ; then
-    [ "$PROVIDE_DHCP" = yes ] && $DHCPD -q -cf /etc/mol/dhcpd-mol.conf $IFACES
+    if [ "$PROVIDE_DHCP" = yes ] ; then
+	touch /var/state/dhcp/dhcpd.leases
+	$DHCPD -q -cf /etc/mol/dhcpd-mol.conf $IFACES
+    fi
     echo 1 > /proc/sys/net/ipv4/ip_forward
 else
     $IPTABLES -t nat -X mol-ns-redirect >& /dev/null
